### ISSUE TITLE
Page scroll to the bottom of page when closing the modal on Safari

### ISSUE THREAD

**Issue by daolf** [algolia/docsearch#1260]
## Description

On Safari, when opening the search modal and then closing it without having made any search, the page scroll to the bottom of the page.

When searching something, it looks like the search box behave nicely.

Safari version: 14.1
Docsearch version: 3.0.0-alpha.42

https://user-images.githubusercontent.com/7622246/149235336-3516931f-a4a7-442d-9072-5462aad88362.mov

## Steps to reproduce

1. Open Safari 14.1
2. Go to [https://www.scrapingbee.com/documentation]
2. Type `cmd + K` to open the modal
3. Press `esc` to close it
4. Window scroll to the bottom of the page

**Live reproduction:**

https://www.scrapingbee.com

## Expected behavior

It should not scroll to the bottom of the page

## Environment

- OS: macOS 11.3.1
- Browser: Safari 14.1
- DocSearch version: 3.0.0-alpha.42

**Comment by shortcuts** [comment]
Hey, are you able to reproduce on our documentation? docsearch.algolia.com

(Sorry I'm not on a computer rn)

**Comment by daolf** [comment]
No, I was not, but it looks like at least one other project had the same issue:
- https://github.com/facebook/docusaurus/pull/4188

I'm instantiating the search in the simplest way:
```
<script src="https://cdn.jsdelivr.net/npm/@docsearch/js@alpha"></script>
<div class="mb-19 pr-15">
    <div id="search" placeholder="Search">
    </div>
</div>
<script>
    window.docsearch({
        // Your Search API Key
        apiKey: '<apiKey>',
        // The index populated by the DocSearch scraper
        indexName: 'scrapingbee_docs',
        // Your Algolia Application ID
        appId: '<APP-ID>',
        // Replace inputSelector with a CSS selector
        // matching your search input
        container: '#search',
        // Set debug to true to inspect the dropdown
        debug: false,
    });
</script>
```

**Comment by daolf** [comment]
Ok so I found out how to fix this.

After noticing that this bug didn't happen on Netlify deploy preview I wanted to understand why.

On Deploy preview mode, Netlify adds a ["feedback"](https://www.netlify.com/products/deploy-previews/) overlay. That overlay seemed to fix the bug.

**Conclusion**

Adding an empty `input` inside a fixed `div` fixes it.

Here is the updated structure of the HTML of our doc page:

```html
<!DOCTYPE html>
<html lang="en">
    {{ partial "head" . }}
    <body>
        <div id="wrapper">
            <div id="content">
            <!--SearchBar-->
            <!--Content-->
            </div>
        </div>
        <!--Magic Fix-->
        <div class="fixed">
            <input type="text">
        </div>
    </body>
</html>
```

Looking at other issues it probably has to do with `tabindex`, like it looks like Safari doesn't like when you close the popup and the browser doesn't have any other input / "tabindexable" element to focus. The Netlify preview feature didn't add an `input`, but an `iframe`. 

But replacing the `iframe` by the empty `input` fixes it 🤷‍♂️

**Comment by shortcuts** [comment]
> Adding an empty input inside a fixed div fixes it.
> The Netlify preview feature didn't add an input, but an iframe.

So anything that fills the DOM along with the DocSearch component should fix this issue?

This is good to know, thanks for investigating, I'll try to reproduce in multiple cases to see if it's something we can handle

**Comment by daolf** [comment]
I don't know if it is anything or a particlular kind element that needs to be focusable by the browser.

**Comment by jackmcdade** [comment]
We're having this issue on https://statamic.dev and haven't been able to find a workaround yet.

**Comment by daolf** [comment]
Hi there, 


I was able to fix this on your dom by adding this just before the end of the `body`

![image](https://user-images.githubusercontent.com/7622246/151199710-f581e5fa-d543-408c-bff9-83eb307d25c7.png)

**Comment by daolf** [comment]
Still have no idea why this works though

**Comment by jackmcdade** [comment]
> Hi there,
> 
> I was able to fix this on your dom by adding this just before the end of the `body`
> 
> ![image](https://user-images.githubusercontent.com/7622246/151199710-f581e5fa-d543-408c-bff9-83eb307d25c7.png)

I tried that but it didn't work for me.

**Comment by daolf** [comment]
Great!

**Comment by mdo** [comment]
I've just encountered this issue as well in trying to update Bootstrap's docs to DocSearch 3. See https://github.com/twbs/bootstrap/pull/35736. The problem can be seen in our staging deploy for that PR: https://deploy-preview-35736--twbs-bootstrap.netlify.app/docs/5.1/getting-started/introduction/.

Anything else we can do to help debug?

**Comment by shortcuts** [comment]
Hey, thanks for coming back to this.

Are you also testing with Safari 14? I'm not able to reproduce with Safari 15.x anymore

**Comment by mdo** [comment]
Yup, still have it in latest Safari 15. I see it whenever I click into it or Cmd+K.

**Comment by shortcuts** [comment]
It seems that [this part](https://github.com/algolia/autocomplete/blob/9640c2d927301e88a4fa77b25d2dfeb7d25b8039/packages/autocomplete-core/src/onKeyDown.ts#L32-L39) is triggered on escape, I'll try to reproduce on the Autocomplete side. (cc @sarahdayan @francoischalifour if you want to double check)

**Comment by sarahdayan** [comment]
Likely related: https://github.com/algolia/autocomplete/issues/592

**Comment by mdo** [comment]
@shortcuts Were you able to investigate this further? Let me know if I can be of help at all :).

**Comment by DanielSGaspar** [comment]
> Ok so I found out how to fix this.
> 
> After noticing that this bug didn't happen on Netlify deploy preview I wanted to understand why.
> 
> On Deploy preview mode, Netlify adds a ["feedback"](https://www.netlify.com/products/deploy-previews/) overlay. That overlay seemed to fix the bug.
> 
> **Conclusion**
> 
> Adding an empty `input` inside a fixed `div` fixes it.
> 
> Here is the updated structure of the HTML of our doc page:
> 
> ```
> <!DOCTYPE html>
> <html lang="en">
>     {{ partial "head" . }}
>     <body>
>         <div id="wrapper">
>             <div id="content">
>             
>             
>             </div>
>         </div>
>         
>         <div class="fixed">
>             <input type="text">
>         </div>
>     </body>
> </html>
> ```
> 
> Looking at other issues it probably has to do with `tabindex`, like it looks like Safari doesn't like when you close the popup and the browser doesn't have any other input / "tabindexable" element to focus. The Netlify preview feature didn't add an `input`, but an `iframe`.
> 
> But replacing the `iframe` by the empty `input` fixes it 🤷‍♂️

As you mention here, the issue is with the focus. So another fix would be to blur the input before closing the modal. This way safari will not try to search for another input to focus and this will also fix the issue.

**Comment by ryanleichty** [comment]
This issue also exists on https://remix.run/. When you're at the top of the page in Safari (v16.6) and open and close the modal with `esc`, it scrolls to the bottom of the page.
